recipe/improve-ffopen-path-resolution.patch

```diff
@@ -0,0 +1,47 @@
+From 50f06fcfe8803ef004b279f548783609bc3bb5ca Mon Sep 17 00:00:00 2001
+From: Matthew Feickert <matthew.feickert@cern.ch>
+Date: Sat, 4 Jan 2025 19:00:25 -0700
+Subject: [PATCH] feat: Make ffopen path resolution more robust
+
+* Use 'trim' to remove trailing spaces and add logic to handle cases where
+  the path ends with a '/'.
+* Remove attempt to for path of '/user/gj/lib/' as this was only valid for
+  Geert Jan van Oldenborgh's machine.
+* Change default path (to later override or change) to '/usr/local/share/ff/'.
+---
+ src/ff/ffinit.f | 16 ++++++----------
+ 1 file changed, 6 insertions(+), 10 deletions(-)
+
+diff --git a/src/ff/ffinit.f b/src/ff/ffinit.f
+index 7602e3c..664dd41 100644
+--- a/src/ff/ffinit.f
++++ b/src/ff/ffinit.f
+@@ -774,19 +774,15 @@
+ *	fullname = 'USR$LOCAL[GEERT]'//name
+ *	open(ifile,file=fullname,status='OLD',READONLY,err=100)
+ *
+-*	first try - my home directory
+-	path = '/user/gj/lib/'
+-	fullname = path(1:index(path,' ')-1)//name
++*	default to the system directory
++	path = '/usr/local/share/ff/'
++
++	fullname = trim(path)//trim(name)
```

Comment:
I don't think Fortran 90 code (like `trim`) can be added to Fortran 77 `.f` files. So will need to get an equivalent to `trim` working.